rclcpp/include/rclcpp/parameter_client.hpp

```diff
@@ -342,9 +342,24 @@ class SyncParametersClient
       qos_profile);
   }
 
-  RCLCPP_PUBLIC
+  template<typename RepT = int64_t, typename RatioT = std::milli>
   std::vector<rclcpp::Parameter>
-  get_parameters(const std::vector<std::string> & parameter_names);
```

Comment:
or maybe simply extend argument with `std::chrono::nanoseconds`?

Reply:
One approach we've used before is to have a "thin" template function which takes a duration, but then just forward that to a protected/private method that requires you provide a nanoseconds argument.

I think that could be good here because it would allow us to keep the implementations in the `.cpp` files. Would you be willing to do that?

For example:

- https://github.com/ros2/rclcpp/blob/62c958be30bc00d48a7b11c347f021ee97ec1911/rclcpp/include/rclcpp/executor.hpp#L245-L255
- https://github.com/ros2/rclcpp/blob/62c958be30bc00d48a7b11c347f021ee97ec1911/rclcpp/include/rclcpp/executor.hpp#L405-L408

Reply:
i thought that also, okay i will do that.